File: x-fly/src/main/java/com/xia/fly/rxhttp/body/upload/ProgressModel.java
package com.xia.fly.rxhttp.body.upload;

import java.io.Serializable;

/**
 * @author xia
 * @date 2018/9/5.
 */
public class ProgressModel implements Serializable {
    //当前读取字节长度
    private long currentBytes;
    //总字节长度
    private long contentLength;
    //是否读取完成
    private boolean done;

    ProgressModel(long currentBytes, long contentLength, boolean done) {
        this.currentBytes = currentBytes;
        this.contentLength = contentLength;
        this.done = done;
    }

    public long getCurrentBytes() {
        return currentBytes;
    }

    public ProgressModel setCurrentBytes(long currentBytes) {
        this.currentBytes = currentBytes;
        return this;
    }

    public long getContentLength() {
        return contentLength;
    }

    public ProgressModel setContentLength(long contentLength) {
        this.contentLength = contentLength;
        return this;
    }

    public boolean isDone() {
        return done;
    }

    public ProgressModel setDone(boolean done) {
        this.done = done;
        return this;
    }

    @Override
    public String toString() {
        return "ProgressModel{" +
                "currentBytes=" + currentBytes +
                ", contentLength=" + contentLength +
                ", done=" + done +
                '}';
    }
}

File: fly-http/src/main/java/com/xia/fly/http/body/upload/ProgressModel.kt
package com.xia.fly.http.body.upload

import java.io.Serializable

/**
 * @author xia
 * @date 2018/9/5.
 */
@Suppress("unused")
class ProgressModel internal constructor(//当前读取字节长度
        private var currentBytes: Long, //总字节长度
        private var contentLength: Long, //是否读取完成
        private var done: Boolean) : Serializable {

    fun getCurrentBytes(): Long {
        return currentBytes
    }

    fun setCurrentBytes(currentBytes: Long): ProgressModel {
        this.currentBytes = currentBytes
        return this
    }

    fun getContentLength(): Long {
        return contentLength
    }

    fun setContentLength(contentLength: Long): ProgressModel {
        this.contentLength = contentLength
        return this
    }

    fun isDone(): Boolean {
        return done
    }

    fun setDone(done: Boolean): ProgressModel {
        this.done = done
        return this
    }

    override fun toString(): String {
        return "ProgressModel{" +
                "currentBytes=" + currentBytes +
                ", contentLength=" + contentLength +
                ", done=" + done +
                '}'.toString()
    }
}
